src/endpoints/authentication/jwt_auth.cpp

```diff
@@ -3,13 +3,94 @@
 
 #include "ccf/endpoints/authentication/jwt_auth.h"
 
+#include "ccf/ds/nonstd.h"
 #include "ccf/pal/locking.h"
 #include "ccf/rpc_context.h"
 #include "ccf/service/tables/jwt.h"
 #include "ds/lru.h"
 #include "enclave/enclave_time.h"
 #include "http/http_jwt.h"
 
+namespace
+{
+  static const std::string multitenancy_indicator{"{tenantid}"};
+  static const std::string microsoft_entra_domain{"login.microsoftonline.com"};
+
+  std::optional<std::string_view> first_non_empty_chunk(
+    const std::vector<std::string_view>& chunks)
+  {
+    for (auto chunk : chunks)
+    {
+      if (!chunk.empty())
+      {
+        return chunk;
+      }
+    }
+    return std::nullopt;
+  }
+
+  bool validate_issuer(
```

Comment:
We should have unit tests for this function.

Reply:
Done